core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2316,37 +2292,37 @@ public void unmount(AlluxioURI alluxioPath)
    *
    * @param inodePath the Alluxio path to unmount, must be a mount point
    * @param journalContext the journal context
-   * @throws InvalidPathException if an invalid path is encountered
+   * @throws InvalidPathException if the given path is not a mount point
    * @throws FileDoesNotExistException if the path to be mounted does not exist
    * @throws IOException if an I/O error occurs
    */
   private void unmountAndJournal(LockedInodePath inodePath, JournalContext journalContext)
       throws InvalidPathException, FileDoesNotExistException, IOException {
-    if (unmountInternal(inodePath.getUri())) {
-      Inode<?> inode = inodePath.getInode();
-      // Use the internal delete API, setting {@code replayed} to true to prevent the delete
+    if (!unmountInternal(inodePath.getUri())) {
+      throw new InvalidPathException("Failed to unmount " + inodePath.getUri() + ". Please ensure"
+          + " the path is an existing mount point and not root.");
+    }
+    try {
+      // Use the internal delete API, setting {@code alluxioOnly} to true to prevent the delete
       // operations from being persisted in the UFS.
-      long fileId = inode.getId();
-      long opTimeMs = System.currentTimeMillis();
-      deleteRecursiveInternal(inodePath, true /* replayed */, opTimeMs,
-          DeleteOptions.defaults().setRecursive(true));
-      DeleteFileEntry deleteFile =
-          DeleteFileEntry.newBuilder().setId(fileId).setRecursive(true).setOpTimeMs(opTimeMs)
-              .build();
-      appendJournalEntry(JournalEntry.newBuilder().setDeleteFile(deleteFile).build(),
-          journalContext);
-      DeleteMountPointEntry deleteMountPoint =
-          DeleteMountPointEntry.newBuilder().setAlluxioPath(inodePath.getUri().toString()).build();
-      appendJournalEntry(JournalEntry.newBuilder().setDeleteMountPoint(deleteMountPoint).build(),
-          journalContext);
+      DeleteOptions deleteOptions =
+          DeleteOptions.defaults().setRecursive(true).setAlluxioOnly(true);
+      deleteAndJournal(inodePath, deleteOptions, journalContext);
+    } catch (DirectoryNotEmptyException e) {
+      // we should never reach here.
+      // deleteAndJournal should never throw DirectoryNotEmptyException when recursive is true.
+      Preconditions.checkState(false);
```

Comment:
instead of `Preconditions.checkState(false);` can you throw a `RuntimeException` with an informative message instead? thanks